llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -44,6 +44,28 @@ constexpr PreprocessorDir PreprocessorDirs[] = {{tgtok::Ifdef, "ifdef"},
                                                 {tgtok::Endif, "endif"},
                                                 {tgtok::Define, "define"}};
 
+// Returns a pointer past the end of a valid macro name at the start of `Str`.
+// Valid macro names match the regular expression [a-zA-Z_][0-9a-zA-Z_]*.
+static const char *lexMacroName(StringRef Str) {
+  assert(!Str.empty());
```

Comment:
There are too many blank lines in this function, which does not match the style in the majority of the code base.

The blank lines give readers a pause, but we don't give a pause for nearly every statement. Variables and the immediate use usually do not need a separator.

Reply:
Makes sense. I have tried to reduce the whitespace, and have 3 sections: initial assert, first char check, rest chech. PTAL.